### PR TITLE
Refactor orchestrator storage and block failure handling

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Refactored orchestrator storage and introduced a new BlockFailure type for improved error handling and recovery.

### What changed?

- Created a new `BlockFailure` type in `internal/common/block_failures.go` with JSON marshaling/unmarshaling functions.
- Refactored the orchestrator storage interface and implementation:
  - Moved storage-related code to `internal/storage/orchestrator/`.
  - Created a new `OrchestratorStorage` interface with specific methods for block failures and latest polled block number.
  - Implemented a `MemoryOrchestratorStorage` that uses an LRU cache.
- Updated `Poller`, `FailureRecoverer`, and `Orchestrator` to use the new storage interface.
- Removed the generic storage connector and memory implementation.

### CODE REVIEW

**Comment by iuwqyir** [comment]
* **#13** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/13?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/12?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by iuwqyir** [comment]
### Merge activity

* **Sep 18, 6:22 AM EDT**: @iuwqyir started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/12).
